### ISSUE TITLE
replace recursive function calls in Participant.js

### ISSUE THREAD

**Issue by d-koppenhagen** [hftl-ims-research/wonder#6]
replace [this part](https://github.com/hftl-ims-research/wonder/blob/wondergui/src/libs/Participant.js#L245-L290) with a simple loop. In my opinion there is no recursive function call necessary.

**Comment by d-koppenhagen** [comment]
the same for this: https://github.com/hftl-ims-research/wonder/blob/wondergui/src/libs/Participant.js#L397-L420

**Comment by d-koppenhagen** [comment]
also check this: https://github.com/hftl-ims-research/wonder/blob/wondergui/src/libs/Participant.js#L285

**Comment by sdruesedow** [comment]
Agree! I think it will simplify a lot, if you propse a better format for the constraints.
